internal/fromproto6/arguments_data.go

```diff
@@ -159,6 +160,289 @@ func ArgumentsData(ctx context.Context, arguments []*tfprotov6.DynamicValue, def
 			}
 		}
 
+		switch parameterWithValidators := parameter.(type) {
+		case function.ParameterWithBoolValidators:
+			for _, functionValidator := range parameterWithValidators.BoolValidators() {
+				boolVal, ok := attrValue.(types.Bool)
+				if !ok {
+					funcError = function.ConcatFuncErrors(funcError, function.NewArgumentFuncError(
+						pos,
+						"Invalid Argument Type: "+
+							"An unexpected error was encountered when converting the function argument from the protocol type. "+
+							"This is always an issue in terraform-plugin-framework used to implement the provider and should be reported to the provider developers.\n\n"+
+							"Please report this to the provider developer:\n\n"+
+							fmt.Sprintf("Expected types.Bool at position %d", pos),
+					))
+
+					continue
+				}
+				req := function.BoolRequest{
+					ArgumentPosition: pos,
+					Value:            boolVal,
+				}
+				resp := &function.BoolResponse{}
+				functionValidator.Validate(ctx, req, resp)
+				if resp.Error != nil {
+					funcError = function.ConcatFuncErrors(funcError, function.NewArgumentFuncError(
+						pos,
+						resp.Error.Error(),
+					))
```

Comment:
Should we have a `continue` statement here so that we only append to the `argumentValues` slice when no error is raised during validation? Similar question for each `case` statement within this `switch`.